tuf/roledb.py

```diff
@@ -900,10 +903,13 @@ def get_delegation_threshold(
   # improperly formatted.
   securesystemslib.formats.NAME_SCHEMA.check_match(repository_name)
 
-  # Raises securesystemslib.exceptions.FormatError,
-  # securesystemslib.exceptions.UnknownRoleError, or
-  # securesystemslib.exceptions.InvalidNameError.
-  _check_rolename(rolename, repository_name)
+  # Does 'rolename' have the correct object format?
+  # This check will ensure 'rolename' has the appropriate number of objects
+  # and object types, and that all dict keys are properly named.
+  tuf.formats.ROLENAME_SCHEMA.check_match(rolename)
+
+  # Raises securesystemslib.exceptions.InvalidNameError.
+  _validate_rolename(rolename)
```

Comment:
A general tip -- NBD here -- that I'd give is to not overgeneralize comments explaining what something does.  In this case, while `check_match` can check the number of objects in something and their types, all it's doing here is making sure that rolename is a string -- it doesn't have objects of various types in it.  So it would be a little better to say that, or to leave it basic, like "check rolename's format".


Try to make commit messages a little more explicit about why a change is being made in the future.  In this case, the why is: the functions retrieve information from the delegating role, and it is not necessary for the delegated-to role to be loaded into roledb for us to retrieve the information, and not always reasonable to expect that the delegated-to role is loaded.

As a totally unrelated matter, note that `_validate_rolename` should really be doing the `check_match` itself. When someone calls it, of course they're interested in that, too, and there's no reason to always call both. Has nothing to do with this PR.